### PR TITLE
feat: router grpc write handler

### PR DESCRIPTION
**Pull request by domodwyer**
Implements the gRPC write service (which previously existed, but returned an error).

This isn't strictly needed at the moment, but can't hurt, and will help the e2e effort (https://github.com/influxdata/influxdb_iox/pull/3905#discussion_r819467597).

---

* feat: grpc write handler (26c43f0a2)

      Implements the gRPC write handler endpoint.

* refactor: enable gRPC handler (7c5ba34d4)

      Plumbs the gRPC write handler into the existing router2 server.

### CODE REVIEW

File: router2/src/server/grpc.rs [hunk 1]
```diff
@@ -1,34 +1,254 @@
...
+            .try_into()
+            .map_err(|e| FieldViolation {
+                field: "database_name".into(),
+                description: format!("Invalid namespace: {}", e),
+            })?;
+
+        let num_tables = tables.len();
+        debug!(
+            num_tables,
+            %namespace,
```
Comment (alamb) [inline router2/src/server/grpc.rs:136]: 💯  for including the namespace here for debugging

File: router2/src/server/grpc.rs [hunk 2]
```diff
@@ -1,34 +1,254 @@
...
+            }),
+        };
+
+        let err = grpc
+            .write(Request::new(req))
+            .await
+            .expect_err("rpc request should fail");
+
+        assert_eq!(err.code(), tonic::Code::NotFound);
+        assert!(err.message().contains("nope"));
```
Comment (alamb) [inline router2/src/server/grpc.rs:252]: `assert_contains!` sometimes is helpful (as it includes the contents of `err.message()` if it fails)

**Review by alamb — APPROVED** [review]
Looks good to me -- getting  an end to end test for this will be nice -- I'll try and add one next week sometime. Traked in https://github.com/influxdata/influxdb_iox/issues/3941

**Comment by alamb** [comment]
Thanks @domodwyer -- this will definitely help

**Review by carols10cents — APPROVED** [review]
LGTM!!

**Pull request merged**